src/sourmash/sourmash_args.py

```diff
@@ -651,10 +654,118 @@ def __init__(self, filename):
     def open(self):
         if self.filename == '-' or self.filename is None:
             return sys.stdout
-        self.fp = open(self.filename, 'w', newline='')
+        if self.filename.endswith('.gz'):
```

Comment:
note to self: this also has wide impact throughout the codebase and should be discussed and tested!!